task/update-deployment/0.1/update-deployment.yaml

```diff
@@ -40,11 +40,13 @@ spec:
           password=$(cat /gitops-auth-secret/password)
           if test -f /gitops-auth-secret/username ; then
             username=$(cat /gitops-auth-secret/username)
+            hostname=$(echo $PARAM_GITOPS_REPO_URL | awk -F/ '{print $3}')
             echo "https://${username}:${password})@${hostname}" > "${HOME}/.git-credentials"
             origin_with_auth=https://${username}:${password}@${remote_without_protocol}.git
           else
             origin_with_auth=https://${password}@${remote_without_protocol}.git
           fi
+           echo -e "[credential \"https://${hostname}\"]\n  helper = store" > "${HOME}/.gitconfig"
```

Comment:
I think this task does auth with this line below
```
git remote set-url origin $origin_with_auth
```
But .git-credentials + .gitconfig does seem like a better approach. Maybe you can remove that line and the `origin_with_auth` variable instead

Reply:
Ah, and I just noticed that the script doesn't create .git-credentials when there's only a password and no username. That would need a fix too

Reply:
well... I don't know exactly how to do this now - I've tried locally to put GH/Gitlab token to .git-credentials without username and it didn't work. 
Any idea?

Reply:
Hmm, I see. I feel like I managed to make that work at some point in the past, let me check.

If not, we can just consider that case unsupported

Reply:
Any luck with trying that? Should we merge as-is?

Reply:
For github, this works:

```bash
echo "https://whatever:$gh_token@github.com/my-org/my-private-repo.git" > ~/.git-credentials
```

But that may be specific to github. Let's just not support the case where the secret only contains a password (except in the update-deployment task, which already supports it so it would be a breaking change).

Reply:
But in that case, let's align the code with the git-clone and prefetch-dependencies tasks to avoid introducing even more inconsistency than we already have